Source/WebCore/platform/graphics/mac/controls/ControlMac.mm

```diff
@@ -242,12 +251,50 @@ static void drawCellInView(const FloatRect& rect, NSCell *cell, NSView *view)
     [cell setControlView:nil];
 }
 
+void ControlMac::drawCellInternal(const FloatRect& rect, float deviceScaleFactor, const ControlStyle& style, NSCell *cell)
+{
+    // For slider cells, draw only the knob.
+    if ([cell isKindOfClass:[NSSliderCell class]]) {
+        [(NSSliderCell *)cell drawKnob:rect];
+        return;
+    }
+
+    if (supportsViewlessCells()) {
+        drawViewlessCell(rect, deviceScaleFactor, style, cell, shouldFlipContext(m_owningPart));
+        return;
+    }
+
+    auto *view = m_controlFactory.drawingView(rect, style);
+    drawCellInView(rect, cell, view);
+}
+
+static void drawViewlessCellFocusRing(const FloatRect& rect, float deviceScaleFactor, const ControlStyle& style, NSCell *cell)
+{
+    applyViewlessCellSettings(deviceScaleFactor, style, cell);
+
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wnonnull"
+    [cell drawFocusRingMaskWithFrame:rect inView:nil];
+#pragma clang diagnostic pop
```

Comment:
Do we have an AppKit radar to allow us to pass `nil` here?

Reply:
Yes. It is rdar://105250010. A FIXME comment was added.